tests/e2e/streams/streams_tests.py

```diff
@@ -342,85 +367,137 @@ def test_check_already_started_stream(topics, connection):
-    assert common.check_one_result_row(
-        cursor, "MATCH (n:VERTEX { id : 42 }) RETURN n")
+    producer.send(topics[0], b"CREATE (n:VERTEX { id : 42 })")
+    assert common.check_one_result_row(cursor, "MATCH (n:VERTEX { id : 42 }) RETURN n")
 
 
 @pytest.mark.parametrize("transformation", TRANSFORMATIONS_TO_CHECK)
 def test_bootstrap_server(producer, topics, connection, transformation):
     assert len(topics) > 0
     cursor = connection.cursor()
     local = "localhost:9092"
-    common.execute_and_fetch_all(cursor,
-                                 "CREATE STREAM test "
-                                 f"TOPICS {','.join(topics)} "
-                                 f"TRANSFORM {transformation} "
-                                 f"BOOTSTRAP_SERVERS \'{local}\'")
+    common.execute_and_fetch_all(
+        cursor,
+        "CREATE STREAM test "
+        f"TOPICS {','.join(topics)} "
+        f"TRANSFORM {transformation} "
+        f"BOOTSTRAP_SERVERS '{local}'",
+    )
     common.start_stream(cursor, "test")
     time.sleep(5)
 
     for topic in topics:
         producer.send(topic, SIMPLE_MSG).get(timeout=60)
 
     for topic in topics:
-        common.check_vertex_exists_with_topic_and_payload(
-            cursor, topic, SIMPLE_MSG)
+        common.check_vertex_exists_with_topic_and_payload(cursor, topic, SIMPLE_MSG)
 
 
 @pytest.mark.parametrize("transformation", TRANSFORMATIONS_TO_CHECK)
 def test_bootstrap_server_empty(producer, topics, connection, transformation):
     assert len(topics) > 0
     cursor = connection.cursor()
     with pytest.raises(mgclient.DatabaseError):
-        common.execute_and_fetch_all(cursor,
-                                     "CREATE STREAM test "
-                                     f"TOPICS {','.join(topics)} "
-                                     f"TRANSFORM {transformation} "
-                                     "BOOTSTRAP_SERVERS ''")
+        common.execute_and_fetch_all(
+            cursor,
+            "CREATE STREAM test "
+            f"TOPICS {','.join(topics)} "
+            f"TRANSFORM {transformation} "
+            "BOOTSTRAP_SERVERS ''",
+        )
+
+
+@pytest.mark.parametrize("transformation", TRANSFORMATIONS_TO_CHECK)
+def test_set_offset(producer, topics, connection, transformation):
+    assert len(topics) > 0
+    cursor = connection.cursor()
+    common.execute_and_fetch_all(
+        cursor,
+        "CREATE STREAM test "
+        f"TOPICS {topics[0]} "
+        f"TRANSFORM {transformation} "
+        "BATCH_SIZE 1",
+    )
+
+    messages = [f"{i} message" for i in range(1, 21)]
+
+    for message in messages:
+        producer.send(topics[0], message.encode()).get(timeout=60)
+
+    def execute_set_offset_and_consume(id):
+        common.execute_and_fetch_all(
+            cursor, f"CALL mg.set_stream_offset('test', {id}) YIELD result"
+        )
+        common.start_stream(cursor, "test")
+
+        time.sleep(2)
```

Comment:
Maybe you can use `check_vertex_exists_with_topic_and_payload` from `common.py`:
`assert check_vertex_exists_with_topic_and_payload(cursor, topics[0], messages[-1])`

It will wait up to 10 seconds and polling the DB for every 0.1 second.

Reply:
I see it won't work when the offset is `-2`. Maybe try to use `check_one_result_row` or `timed_wait` from `common.py`, both of them should suit this scenario with a proper query.

Reply:
I am unsure how `timed_wait` here would help and I don't see how `check_one_result_row` makes this any simpler?

Reply:
I think `assert check_one_result_row(cursor, f"MATCH (n: MESSAGE {{payload: {messages[-1]}}}) RETURN n")` would do the trick: it might finish faster than 2 seconds if the stream is fast, but would wait up to 10 seconds. In is more robust in case of random slowdowns etc.